src/main/java/org/prebid/server/bidder/adocean/AdoceanBidder.java

```diff
@@ -206,59 +204,67 @@ private String buildUrl(String impId, ExtImpAdocean extImpAdocean, String consen
         final Integer test = bidRequest.getTest();
         final String resolvedUrl = resolveEndpointUrl(extImpAdocean, test);
 
-        final URIBuilder uriBuilder;
+        final UriTemplate uriTemplate;
         try {
-            uriBuilder = new URIBuilder(resolvedUrl);
-        } catch (URISyntaxException e) {
+            final String paramsTemplate = resolvedUrl.contains("?") ? "{&queryParams*}" : "{?queryParams*}";
```

Comment:
I am seeing this type of code in almost each class. You can wrap Vertx `UriTemplateImpl` in custom `UriTemplate` implementation and hide it, or create util method or smth.